daemons/controld/controld_te_utils.c

```diff
@@ -269,7 +269,7 @@ tengine_stonith_notify(stonith_t * st, stonith_event_t * st_event)
                st_event->origin, st_event->id);
 
     if (st_event->result == pcmk_ok) {
-        crm_node_t *peer = crm_find_peer_full(0, st_event->target, CRM_GET_PEER_ANY);
+        crm_node_t *peer = crm_find_known_peer_full(0, st_event->target, CRM_GET_PEER_ANY);
```

Comment:
tengine_stonith_notify() is called on all nodes, whereas do_pe_invoke_callback() is called only on the DC. I think that means this fix will only work on a node that has been DC when the target was in the CIB.

Reply:
That makes sense now.